torch/distributed/_composable/replicate.py

```diff
@@ -85,9 +85,12 @@ def init(
         device_mesh = kwargs.get("device_mesh", None)
         self.module = module
         ignored_params = {p for m in ignored_modules for p in m.parameters()}
+        for submodule in module.modules():
+            if _is_fully_sharded(submodule):
+                ignored_params.update(submodule.parameters())
```

Comment:
This is quadratic in the number of FSDP parameters, but this makes the code much simpler :/

Reply:
looks like they are all excuted on cpu time and it's one time overhead during init ?  
should I copy this PR and add unit test to cover `replicate(fully_shard)`?  I can also raach out to Chien-chin for review. just want to save your time but let me know if you're interested in handling it

Reply:
yea I don't think this is a real concern